homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,243 @@
+"""Expose Synology DSM as a media source."""
+from __future__ import annotations
+
+import mimetypes
+from pathlib import Path
+import tempfile
+
+from aiohttp import web
+from synology_dsm.api.photos import SynoPhotosAlbum, SynoPhotosItem
+from synology_dsm.exceptions import SynologyDSMException
+
+from homeassistant.components import http
+from homeassistant.components.media_player import MediaClass
+from homeassistant.components.media_source import (
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+    Unresolvable,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+from .models import SynologyDSMData
+
+
+async def async_get_media_source(hass: HomeAssistant) -> MediaSource:
+    """Set up Synology media source."""
+    # Synology photos support only a single config entry
+    entries = hass.config_entries.async_entries(DOMAIN)
+    return SynologyPhotosMediaSource(hass, entries)
+
+
+@callback
+def async_setup(hass: HomeAssistant) -> None:
+    """Set up local media source."""
+    hass.http.register_view(SynologyDsmMediaView(hass))
```

Comment:
as mentioned above, do this register just withing the `async_get_media_source()`